counters.hh

```diff
@@ -253,23 +252,31 @@ protected:
 private:
     class shard_iterator {
     public:
-        using iterator_category = std::input_iterator_tag;
+        using iterator_category = std::forward_iterator_tag;
         using value_type = basic_counter_shard_view<is_mutable>;
         using difference_type = std::ptrdiff_t;
         using pointer = basic_counter_shard_view<is_mutable>*;
-        using reference = basic_counter_shard_view<is_mutable>&;
+        using reference = std::conditional_t<is_mutable == mutable_view::yes,
+            basic_counter_shard_view<is_mutable>&,
+            const basic_counter_shard_view<is_mutable>&>;
     private:
         managed_bytes_basic_view<is_mutable> _current;
         basic_counter_shard_view<is_mutable> _current_view;
         size_t _pos = 0;
     public:
+        shard_iterator() noexcept = default;
```

Comment:
With this change `_current_view` is _not_ constructed with `_current`.
It is not the same as-if before the change:
```cpp
shard_iterator(managed_bytes_basic_view<mutable_view::yes>(), 0) // or mutable_view::no
```

Why do we need default empty constructor for?

Reply:
Iterator concepts require it, but it doesn't need to do anything.